sessionctx/variable/session.go

```diff
@@ -2389,3 +2396,12 @@ func (s *SessionVars) GetSeekFactor(tbl *model.TableInfo) float64 {
 	}
 	return s.seekFactor
 }
+
+// IsRcCheckTsRetryable checks if the current error is retryable for `RcReadCheckTS` path.
+func (s *SessionVars) IsRcCheckTsRetryable(err error) bool {
+	if err == nil {
+		return false
+	}
+	// The `RCCheckTS` flag of `stmtCtx` is reset d
+	return s.RcReadCheckTS && s.StmtCtx.RCCheckTS && errors.ErrorEqual(err, kv.ErrWriteConflict)
```

Comment:
We reuse the error code kv.ErrWriteConflict, does it affect the write sqls behavior when write-conflict occur really.

Reply:
May WriteConflict error be met during pessimistic lock phase when executing a DML statement, and then checked by this function?

Reply:
I've added another condition judgement checking if it's read only statement.